tests/IndexerTest.php

```diff
@@ -0,0 +1,196 @@
+     */
+    public function testAliasGetsCreated(): void
+    {
+        $elasticClient = new ClientStub();
+        $indexer = $this->createInstance($elasticClient);
+
+        $indexer->create(new HandlerStub());
+
+        self::assertSame(['valid_new'], $elasticClient->getCreatedAliases());
+    }
+
+    /**
+     * Ensure the cleaning process only disregards indices unrelated to search handlers
+     *
+     * @return void
+     */
+    public function testCleaningIndicesDoesNotRemoveUnrelatedIndices(): void
+    {
+        // #1. Have random indices exist (and aliases?)
+        // #2. Have search handler indices exi
+        $client = new ClientStub(
+            null,
+            null,
+            [['name' => 'unrelated-index'], ['name' => 'irrelevant-index'], ['name' => 'valid-123']]
+        );
+        $indexer = $this->createInstance($client);
+
+        $indexer->clean([new HandlerStub()]);
+
+        // unrelated-index and irrelevant-index should not be touched, because they are unrelated to search handlers
+        self::assertSame(['valid-123'], $client->getDeletedIndices());
+    }
+
+    /**
+     * Ensure the cleaning process only cares about indices that are related to search handlers
+     *
+     * @return void
+     */
+    public function testCleaningIndicesRepectsIndicesFromAliases(): void
+    {
+        // #1. Have random indices exist (and aliases?)
+        // #2. Have search handler indices exi
+        $client = new ClientStub(
+            null,
+            null,
+            [['name' => 'unrelated-index'], ['name' => 'valid-unused']],
+            [['index' => 'valid', 'name' => 'anything']]
+        );
+        $indexer = $this->createInstance($client);
+
+        $indexer->clean([new HandlerStub()]);
+
+        self::assertSame(['valid-unused'], $client->getDeletedIndices());
+    }
+
+    /**
+     * Ensure the swap method removes the _new alias
+     *
+     * @return void
+     */
+    public function testIndexSwapperRemovesNewAlias(): void
+    {
+        $elasticClient = new ClientStub(
+            true,
+            null,
+            null,
+            [['name' => 'valid_new', 'index' => 'valid_201900502']]
+        );
+        $indexer = $this->createInstance($elasticClient);
+
+        $indexer->indexSwap(new HandlerStub());
+
+        self::assertSame(['valid_new'], $elasticClient->getDeletedAliases());
+    }
+
+    /**
+     * Ensure the index<->alias swap does indeed happen
+     *
+     * @return void
+     */
+    public function testIndexSwapperSwapsAlias(): void
+    {
+        $elasticClient = new ClientStub(
+            true,
+            null,
+            null,
+            [['name' => 'valid_new', 'index' => 'valid_201900502']]
+        );
+        $indexer = $this->createInstance($elasticClient);
+
+        $indexer->indexSwap(new HandlerStub());
+
+        self::assertSame(['valid'], $elasticClient->getSwappedAliases());
```

Comment:
I'd like this to test what valid actually got swapped to, plus all these tests need their expected in arrange